### PR TITLE
[`FA`] Fix some model tests

### PR DESCRIPTION
**Pull request by vasqu**
As internally discussed we want to have FA in our CIs so the first step is to enable it to run without corrupting the cuda state (i.e. device errors trigger subsequent device errors if in the same process).

cc @ydshieh @Cyrilvallez

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
Thank you. LGTM but not a FA2 expert :-)

File: src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py [hunk 1]
```diff
@@ -221,12 +221,8 @@ def forward(
 @auto_docstring
 class ModernBertDecoderPreTrainedModel(ModernBertPreTrainedModel):
     config: ModernBertDecoderConfig
-    base_model_prefix = "model"
     _skip_keys_device_placement = ["past_key_values"]
     _no_split_modules = ["ModernBertDecoderLayer"]
-    _supports_flash_attn = True
-    _supports_sdpa = False
-    _supports_gradient_checkpointing = True
```
Comment (vasqu) [inline src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py:229]: Flags are already in parent

File: src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py [hunk 2]
```diff
@@ -280,6 +276,20 @@ def init_weight(module: nn.Module, std: float):
...
+        """We overwrite this to make sdpa the first selection again if nothing was requested."""
+
+        try:
+            attn_implementation = (
+                "sdpa" if attn_implementation is None and self._sdpa_can_dispatch() else attn_implementation
+            )
+        except (ValueError, ImportError):
+            pass
+
+        return super()._check_and_adjust_attn_implementation(attn_implementation, is_init_check)
```
Comment (vasqu) [inline src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py:291]: This was the easiest solution, thought about changing the test prep but that seems more faulty

File: src/transformers/models/siglip/modeling_siglip.py [hunk 3]
```diff
@@ -649,7 +648,10 @@ def forward(
 
         # note: SigLIP's text model does not use a causal mask, unlike the original CLIP model.
         # expand attention_mask
-        if attention_mask is not None and not self._use_flash_attention_2:
+        uses_flash_attention = "flash" in self.config._attn_implementation
+        if uses_flash_attention:
+            attention_mask = None
+        elif attention_mask is not None and not uses_flash_attention:
```
Comment (vasqu) [inline src/transformers/models/siglip/modeling_siglip.py:654]: Quick fix here, no mask is used so we can ignore it

File: tests/test_modeling_common.py [hunk 4]
```diff
@@ -3501,6 +3501,11 @@ def flash_attn_inference_equivalence(self, attn_implementation: str, padding_sid
                 model = model_class.from_pretrained(tmpdirname, torch_dtype=torch.bfloat16)
                 model.to(torch_device)
 
+                # Some models have support for FA but not SDPA - making sure we have a valid attention
+                initial_attention_implementation = "sdpa"
+                if model.config._attn_implementation != "sdpa":
+                    initial_attention_implementation = "eager"
```
Comment (vasqu) [inline tests/test_modeling_common.py:3507]: This makes it more robust

File: src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py (hunk 2 above)
Comment (Cyrilvallez) [inline src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py:291]: I think modernbert is supposed to use fa2 by default no??

**Review by Cyrilvallez — COMMENTED** [review]
A bit late, but modernbert uses fa2 by default by design! This would be quite breaking

File: src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py (hunk 2 above)
Comment (ydshieh) [reply src/transformers/models/modernbert_decoder/modeling_modernbert_decoder.py:291]: oops, @vasqu merged with only my approval 🥶
